syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service LoanPaymentService {
  rpc GetAll (GetAllRequest) returns (GetAllLoanPaymentsResponse);
  rpc GetById (GetLoanPaymentByIdRequest) returns (LoanPaymentModel);
  rpc GetByLoan (GetLoanPaymentsByLoanRequest) returns (GetAllLoanPaymentsResponse);
}

message LoanPaymentModel {
  int32 payment_id = 1;
  optional int32 loan_id = 2;
  optional string amount = 3;
  optional DateOnly payment_date = 4;
  optional bool is_paid = 5;
}

message GetAllLoanPaymentsResponse {
  repeated LoanPaymentModel loan_payments = 1;
}

message GetLoanPaymentByIdRequest {
  int32 payment_id = 1;
}

message GetLoanPaymentsByLoanRequest {
  int32 loan_id = 1;
}
